pilot/pkg/bootstrap/server.go

```diff
@@ -1101,11 +1105,27 @@ func (s *Server) maybeCreateCA(caOpts *caOptions) error {
 			if s.RA, err = s.createIstioRA(s.kubeClient, caOpts); err != nil {
 				return fmt.Errorf("failed to create RA: %v", err)
 			}
+			s.environment.AddMeshHandler(func() {
+				meshConfig := s.environment.Mesh()
+				caCertificates := meshConfig.CaCertificates
+				if len(caCertificates) == 0 {
+					return
+				}
+				s.RA.SetCACertificatesFromMeshConfig(caCertificates)
+			})
 		}
 		if !s.isDisableCa() {
 			if s.CA, err = s.createIstioCA(corev1, caOpts); err != nil {
 				return fmt.Errorf("failed to create CA: %v", err)
 			}
+			s.environment.AddMeshHandler(func() {
+				meshConfig := s.environment.Mesh()
+				caCertificates := meshConfig.CaCertificates
+				if len(caCertificates) == 0 {
+					return
+				}
+				s.CA.SetCACertificatesFromMeshConfig(caCertificates)
```

Comment:
Given that this logic is quite specific to how Istiod functions as an RA, we should keep the logic out of the CA code as much as possible IMO

Reply:
Thanks! This should not be needed

Reply:
remvoed